pinot-core/src/main/java/org/apache/pinot/core/common/datatable/DataTableImplV2V3.java

```diff
@@ -344,6 +395,20 @@ public void addException(ProcessingException processingException) {
     return byteArrayOutputStream.toByteArray();
   }
 
+  private byte[] serializePositionalData()
+      throws IOException {
+    ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+    DataOutputStream dataOutputStream = new DataOutputStream(byteArrayOutputStream);
+
+    dataOutputStream.writeInt(_positionalData.length);
+    for (String entry : _positionalData) {
+      byte[] bytes = StringUtil.encodeUtf8(entry);
```

Comment:
Some comments on the format here would be useful. We don't write the enum constant itself. Just the value (length+bytes) corresponding to the ordinal/position of the constant. Correct ?

Reply:
yes, your understanding is correct.